session/manager.go

```diff
@@ -51,65 +59,139 @@ func (m *Manager) BuildSession(handler sessioncontract.Driver, sessionID ...stri
+		}
+
+		m.drivers[driverName] = driverInstance
+		m.startGcTimer(driverInstance) // Pass the concrete instance
+		return driverInstance, nil
 	}
 
-	return m.drivers[driverName], nil
+	if driverName == "file" {
+		lifetime := m.config.GetInt("session.lifetime")
+		fileDriver := driver.NewFile(m.config.GetString("session.files"), lifetime)
+		driverInstance = fileDriver
+		m.drivers[driverName] = driverInstance
+		m.startGcTimer(driverInstance)
+		return driverInstance, nil
+	}
+
+	return nil, errors.SessionDriverNotSupported.Args(driverName)
 }
 
-func (m *Manager) Extend(driver string, handler func() sessioncontract.Driver) error {
-	if m.drivers[driver] != nil {
+// Extend registers a factory function for a given driver name.
+func (m *Manager) Extend(driver string, factory func() contractssession.Driver) error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if _, exists := m.factories[driver]; exists {
 		return errors.SessionDriverAlreadyExists.Args(driver)
 	}
-	m.drivers[driver] = handler()
-	m.startGcTimer(m.drivers[driver])
+	m.factories[driver] = factory
 	return nil
 }
 
-func (m *Manager) ReleaseSession(session sessioncontract.Session) {
+func (m *Manager) ReleaseSession(session contractssession.Session) {
 	session.Flush().
 		SetDriver(nil).
 		SetName("").
 		SetID("")
 	m.sessionPool.Put(session)
 }
 
-func (m *Manager) acquireSession() sessioncontract.Session {
-	session := m.sessionPool.Get().(sessioncontract.Session)
+func (m *Manager) acquireSession() contractssession.Session {
+	session := m.sessionPool.Get().(contractssession.Session)
 	return session
 }
 
+// getDefaultDriver reads the default driver name from config.
 func (m *Manager) getDefaultDriver() string {
 	return m.config.GetString("session.driver")
 }
 
-func (m *Manager) extendDefaultDrivers() {
-	if err := m.Extend("file", m.createFileDriver); err != nil {
-		panic(errors.SessionDriverExtensionFailed.SetModule(errors.ModuleSession).Args("file", err))
+// registerConfiguredDrivers ONLY reads from config.
+func (m *Manager) registerConfiguredDrivers() {
+	configuredDrivers := m.config.Get("session.drivers", map[string]any{})
+	driversMap, ok := configuredDrivers.(map[string]any)
+	if !ok {
+		return
 	}
-}
+	for name, driverConfigAny := range driversMap {
+		driverConfig, ok := driverConfigAny.(map[string]any)
+		if !ok {
+
+			continue
+		}
+
+		viaFactoryAny, exists := driverConfig["via"]
+		if !exists {
+
+			continue
+		}
+
+		// Factory from config: func() (contractssession.Driver, error)
+		viaFactoryWithError, ok := viaFactoryAny.(func() (contractssession.Driver, error))
+		if !ok {
+			continue
+		}
 
-func (m *Manager) createFileDriver() sessioncontract.Driver {
-	lifetime := m.config.GetInt("session.lifetime")
-	return driver.NewFile(m.config.GetString("session.files"), lifetime)
+		// Wrapper for Extend signature: func() contractssession.Driver
+		factoryWrapper := func() contractssession.Driver {
```

Comment:
The error should be returned instead of `color.Errorf`.

```suggestion
		factoryWrapper := func() (contractssession.Driver, error) {
```